src/rdmc.py

```diff
@@ -100,7 +100,7 @@
     # enable fips mode if our special functions are available in _ssl and OS is
     # in FIPS mode
     FIPSSTR = ""
-    if Encryption.check_fips_mode_os() and not ssl.FIPS_mode():
+    if Encryption.check_fips_mode_os() and not Encryption.check_fips_mode_ssl:

```

Comment:
is this Encryption.check_fips_mode_ssl() with brackets?

Reply:
It should be with brackets.....

Also note that this is only for 1.0.2, 1.1.0, 1.1.1 series of OpenSSL.

Openssl v3 has removed the FIPS_MODE call.

Ideally this python library must not assert or change FIPS mode on Linux. Since when linux kernel is in fips mode, python and openssl are too. And these code paths are only executed on Linux, thus maybe just drop them all, as that will mean things work correctly in FIPS mode on Linux with any current, past or future FIPS OpenSSL modules.